src/Analyzers/CSharp/Tests/RemoveUnnecessaryCast/RemoveUnnecessaryCastTests.cs

```diff
@@ -627,18 +644,7 @@ public static void Main()
     {
         int b = 5;
 
-        long f1 = (b == 5) ? 4 : [|(long)|]5;
-    }
-}";
-            var batchFixedSource =
-@"
-class Test
-{
-    public static void Main()
-    {
-        int b = 5;
-
-        long f1 = (b == 5) ? 4 : 5;
+        long f1 = (b == 5) ? 4 : (long)5;
```

Comment:
this is a regression.  it is tracked in https://github.com/dotnet/roslyn/issues/56938.  The issue here is that removing the cast does change runtime semantics.  Basically teh widening step gets moved *outside* of the conditional.  This will have no observable impact, but it does violate the rule that we want the runtime operations to stay the same. 

So this sort of 'moving' case will have to be special cased in a followup PR.